renku/service/jobs/datasets.py

```diff
@@ -22,59 +22,47 @@
 from renku.core.errors import DatasetExistsError, ParameterError
 from renku.core.management.datasets import DownloadProgressCallback
 from renku.core.utils.contexts import chdir
-from renku.service.jobs.constants import USER_JOB_STATE_COMPLETED, \
-    USER_JOB_STATE_FAILED, USER_JOB_STATE_IN_PROGRESS
-from renku.service.serializers.jobs import UserJob
-from renku.service.utils import make_project_path, repo_sync
+from renku.service.cache.models.job import USER_JOB_STATE_COMPLETED, \
+    USER_JOB_STATE_IN_PROGRESS
+from renku.service.cache.serializers.job import JobSchema
+from renku.service.utils import repo_sync
 from renku.service.views.decorators import requires_cache
 
 
-def fail_job(cache, user_job, user, error):
-    """Mark job as failed."""
-    user_job['state'] = USER_JOB_STATE_FAILED
-
-    if 'extras' not in user_job:
-        user_job['extras'] = {}
-
-    user_job['extras']['error'] = error
-    cache.set_job(user, user_job)
-
-
 class DatasetImportJobProcess(DownloadProgressCallback):
     """Track dataset import job progress."""
 
-    schema = UserJob()
+    schema = JobSchema()
 
-    def __init__(self, cache, user, job):
+    def __init__(self, cache, job):
         """Construct dataset import job progress."""
         self.cache = cache
-        self.user = user
         self.job = job
 
         super().__init__(None, None)
 
     def __call__(self, description, total_size):
         """Job progress call."""
-        self.job['state'] = USER_JOB_STATE_IN_PROGRESS
-        self.job['extras'] = {
+        self.job.state = USER_JOB_STATE_IN_PROGRESS
```

Comment:
This could be put in a method on job, to not expose the internal of the job state to the outside world. In line with `Job.fail_job()`